pkgs/development/python-modules/trino-python-client/default.nix

```
@@ -0,0 +1,77 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+  # build
+, setuptools
+  # required
+, pytz
+, requests
+, tzlocal
+  # optional
+, requests-kerberos
+, sqlalchemy
+, keyring
+  # tests
+, pytestCheckHook
+, httpretty
+, pytest-runner
+}:
+
+buildPythonPackage rec {
+  pname = "trino-python-client";
+  version = "0.322.0";
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    repo = pname;
+    owner = "trinodb";
+    rev = "refs/tags/${version}";
+    hash = "sha256-Hl88Keavyp1QBw67AFbevy/btzNs7UlsKQ93K02YgLM=";
+  };
+
+  nativeBuildInputs = [
+    setuptools
+  ];
+
+  propagatedBuildInputs = [
+    pytz
+    requests
+    tzlocal
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    httpretty
+    pytest-runner
+    requests-kerberos
+    sqlalchemy
+  ];
```

It would be semantically more correct to reference the optional-dependency groups here. Also, I doubt the project really needs `pytest-runner`, which has been deprecated for years.

```suggestion
  nativeCheckInputs = [
    httpretty
    pytestCheckHook
  ] ++ passthru.optional-dependencies.all;
```